core/src/test/java/com/linecorp/armeria/server/ProxyServerAuthorityHeaderTest.java

```diff
@@ -0,0 +1,128 @@
+            });
+            sb.service("/toHttp2", (ctx, req) -> {
+                backendHeaders.set(req.headers());
+                return HttpResponse.of(200);
+            });
+        }
+    };
+
+    @RegisterExtension
+    static final ServerExtension proxy = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.service("/toHttp1", (ctx, req) -> {
+                proxyHeaders.set(req.headers());
+                final WebClient client = WebClient.of(backend.uri(SessionProtocol.H1C));
+                return client.execute(req);
+            });
+            sb.service("/toHttp2", (ctx, req) -> {
+                proxyHeaders.set(req.headers());
+                final WebClient client = WebClient.of(backend.uri(SessionProtocol.H2C));
+                return client.execute(req);
+            });
+        }
+    };
+
+    @BeforeEach
+    void setUp() {
+        backendHeaders.set(null);
+        proxyHeaders.set(null);
+    }
+
+    @Test
+    void http1ToHttp1() {
+        final WebClient client = WebClient.of(proxy.uri(SessionProtocol.H1C));
+        client.get("/toHttp1").aggregate().join();
+        final RequestHeaders proxyRequestHeaders = proxyHeaders.get();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.HOST)).isNotNull();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNull();
+
+        final RequestHeaders backendRequestHeaders = backendHeaders.get();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.HOST)).isNotNull();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNull();
+    }
+
+    @Test
+    void http2ToHttp2() {
+        final WebClient client = WebClient.of(proxy.uri(SessionProtocol.H2C));
+        client.get("/toHttp2").aggregate().join();
+        final RequestHeaders proxyRequestHeaders = proxyHeaders.get();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.HOST)).isNull();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNotNull();
+
+        final RequestHeaders backendRequestHeaders = backendHeaders.get();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.HOST)).isNull();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNotNull();
+    }
+
+    @Test
+    void http1ToHttp2() {
+        final WebClient client = WebClient.of(proxy.uri(SessionProtocol.H1C));
+        client.get("/toHttp2").aggregate().join();
+        final RequestHeaders proxyRequestHeaders = proxyHeaders.get();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.HOST)).isNotNull();
+        assertThat(proxyRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNull();
+
+        final RequestHeaders backendRequestHeaders = backendHeaders.get();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.HOST)).isNull();
+        assertThat(backendRequestHeaders.get(HttpHeaderNames.AUTHORITY)).isNotNull();
```

Comment:
It seems like HAproxy does not translate the `Host` header to `:authority` header when it's acting as a proxy between `H1` and `H2`. (However, we do.)
https://github.com/haproxy/haproxy/issues/716#issuecomment-650733053

Reply:
+1 for HAProxy behavior so that we allow a user to build a proxy server that does not touch the original request.

Reply:
Fixed. 😄